packages/steiger-plugin-fsd/src/no-public-api-sidestep/index.ts

```diff
@@ -64,12 +65,30 @@ const noPublicApiSidestep = {
             continue
           }
 
-          const index = getIndex(segment)
-          if (resolvedDependency !== index?.path) {
-            diagnostics.push({
-              message: `Forbidden sidestep of public API when importing from "${dependency}".`,
-              location: { path: sourceFile.file.path },
-            })
+          const segmentIndex = getIndex(segment)
+          if (segment.type === 'folder' && resolvedDependency !== segmentIndex?.path) {
+            if (dependencyLocation.layerName === 'shared' && ['ui', 'lib'].includes(dependencyLocation.segmentName)) {
```

Comment:
Non-blocking: this set of conditions is pretty difficult to read. I would create several variables that could make it possible to read it almost like plain English e.g. `isSharedLayer`, `isUIorLib` etc. What do you think

Reply:
Yeah, I also thought that, but didn't know how to improve. I'll try your suggestion

Reply:
Decided to go with this in the end:
```ts
const { layerName, segmentName } = dependencyLocation
if (layerName === 'shared' && ['ui', 'lib'].includes(segmentName)) {}
```

Reply:
That works for me, thanks!